elastic/apm-server
# beater/jaeger: introduce "auth_tag" configuration

## Motivation/summary

We introduce support for configuring the server to look for a specified Process tag in Jaeger batches, and use the value for authorization. The value is expected to have the same format as HTTP authorization headers that the Elastic APM agents support. By default Jaeger authorization is disabled, as there is no standard auth tag name.

This enables token-based auth as an alternative to client certificate auth.

Note for reviewers: an alternative approach to configuration would be to have a hard-coded auth tag, like "elastic_auth", and have that enabled by default (or enabled with a boolean). Since token-based auth isn't a native feature of Jaeger clients/agents, I thought it might be a bit non-intuitive, hence why I made it opt-in. Let me know what you think.

## Checklist


## How to test these changes

1. Configure `apm-server.jaeger.auth_tag: "authorization"`
1. Configure `apm-server.secret_token: "foo"`
1. Run Jaeger Agent without any tags, run HotROD, and check the agent shows "not authorized" errors and nothing is indexed
1. Run Jaeger Agent with `--agent.tags authorization=foo`, run HotROD
1. Check it all works, and there is no "labels.authorization" field in any docs

## Related issues

Closes #3339
